tutorials/ardu-pi-serial-part-1/index.md

```diff
@@ -0,0 +1,330 @@
+  pinMode(LED_BUILTIN, OUTPUT);
+  digitalWrite(LED_BUILTIN, LOW);
+}
+
+void loop(){
+  if (Serial.available()) {
+     ch = Serial.read();
+     if ( ch == '0' ) {
+       sendTelemetry();
+     }
+  }
+}
+
+void sendTelemetry() {
+  digitalWrite(LED_BUILTIN, HIGH);
+  memset(sendBuffer, 0, sizeof(sendBuffer));
+  int windgen = analogRead(WINDGEN_PIN);
+  int solargen = analogRead(SOLARGEN_PIN);
+  sprintf(sendBuffer, "S s:%d w:%d", solargen, windgen);
+  Serial.println(sendBuffer);
+  digitalWrite(LED_BUILTIN, LOW);
+}
+```
+
+1. Select `File > Save As...` and save your sketch with a suitable name such as `ardu-pi-serial`
+
+**Figure 4.** *Arduino IDE and Sketch*
+![Sketch diagram][sketchdiag]
+
+[sketchdiag]: sketch.png
+
+### Test reading Arduino sensors over serial connection from your machine
+The Arduino Sketch client has the following functionality:
+
+- Check if the serial connection has data available to read
+- If the received character (command) is 0:
+    - Read the voltage values of pins A0 and A1
+    - Print the sensor values to the serial connection
+
+Perform the following steps to deploy and test the Arduino sketch:
+
+1. Select `Sketch > Verify/Compile` from the Arduino IDE menu. If the code is successfully verified, the Arduino IDE should display "Done compiling".  You should also see output similar to the following:
+```
+Sketch uses 4708 bytes (14%) of program storage space. Maximum is 32256 bytes.
+Global variables use 249 bytes (12%) of dynamic memory, leaving 1799 bytes for local variables. Maximum is 2048 bytes.
+```
+1. Select `Sketch > Upload` to transfer the compiled Sketch to your Arduino over the USB serial connection. You should see the Arduino's LEDs blink rapidly while the binary is transferred over.
+1. Select `Tools > Serial Monitor` to open the Serial Monitor window.
+1. Ensure that the baud rate is set to "115200 baud".
+1. Test requesting the current sensor values from the Arduino, by typing `0` (zero) in the Serial Monitor's input window, and pressing `<enter>`. If the Sketch and serial connection works, you should see a response similar to: `S s:136 w:43` in the Serial Monitor.
+
+Here, `S` indicates sensor output, `s:136` is the current value of the solar panel (analog pin `A1`) and `w:43` is the current value of the wind generator / DC motor (analog pin `A0`). This is how the Python client app on the Pi will request the sensor readings from the Arduino later on. Command `0` is an example of a simple protocol used to command an Arduino program.
+
+Different inputs (e.g `1`) could be used to command the Arduino to perform a different action, such as turning a servo motor. In this tutorial, the Arduino Sketch implements a single command `0`, which reads the voltage values of analog pins `A0` and `A1`.
+
+- Unplug the USB cable from your local machine, to disconnect and power down the Arduino. The Arduino will now run the sketch automatically, when ever it is connected to a power source.
```

Comment:
Strike the comma after the word "machine".

Reply:
Done